python/tvm/relay/frontend/mxnet.py

```diff
@@ -876,6 +888,142 @@ def _lstm_cell(data, states, i2h_weight, h2h_weight, i2h_bias, h2h_bias):
+
+    new_body = _expr.bind(loop_body.body, binds)
+    loop_body = _expr.Function(
+        list(set(analysis.free_vars(new_body)) - set(params)),
+        new_body,
+        loop_body.ret_type,
+        loop_body.type_params,
+        loop_body.attrs)
+
+    # relay.loops.foreach is typed and therefore stricter then the
+    # interface MxNet exposes.
+    #
+    # In order to handle this we check two cases and normalize the
+    # appropriate code.
+    #
+    # When returning zero states we will tuple the singleton
+    # body.
+    if len(prev_states) == 0:
+        tupled_body = _expr.Tuple([loop_body.body])
+        loop_body = _expr.Function(
+            loop_body.params,
+            tupled_body, None,
+            loop_body.type_params,
+            loop_body.attrs)
+
+    # In the case where the output data is ignored, we will pass
+    # it along as normal.
+    if num_out_data == 0:
+        unit = _expr.Tuple([])
+        tupled_body = _expr.Tuple([unit, loop_body.body])
+        loop_body = _expr.Function(
+            loop_body.params,
+            tupled_body, None,
+            loop_body.type_params,
+            loop_body.attrs)
+
+    # Finally we will invoke `relay.loops.foreach` with the data,
+    # loop_body, and then initial states.
+    loop = loops.foreach(data[0], loop_body, prev_states, mod=mod)
+
+    loop_var = _expr.GlobalVar('foreach_loop')
+
+    # Bind the parameters outside of the loop,
+    # the final function will take the data to iterate over
+    # the state, and the parameters.
+    inner_params = []
+    binds = {}
+    for param in params:
+        inner_param = _expr.Var(param.name_hint, param.type_annotation)
+        binds[param] =  inner_param
+        inner_params.append(inner_param)
+
+    new_body = _expr.bind(loop.body, binds)
+    loop = _expr.Function(
+        [*loop.params, *inner_params],
+        new_body,
+        loop.ret_type,
+        loop.type_params,
+        loop.attrs)
+
+    # We insert the generated loop into the module.
+    mod[loop_var] = loop
+
+    # We return a tuple with a list of outputs (1), and
+    # the number of previous states.
+    start_loop = loop_var(data[0], *(prev_states + params))
+
+    # Now we can just drop the data.
+    if num_out_data == 0:
+        return _expr.Tuple([_expr.TupleGetItem(start_loop, 1)])
+
+    return _expr.TupleWrapper(start_loop, 1 + len(prev_states))
+
+
+def _mx_while_loop(inputs, attrs, subgraphs_json, dtype_info, mod):
+    cond_json, loop_body_json = subgraphs_json
+    cond = _mx.symbol.load_json(json.dumps(cond_json))
+    loop_body = _mx.symbol.load_json(json.dumps(loop_body_json))
+
+    cond_input_locs = attrs.get_int_tuple("cond_input_locs")
+    func_input_locs = attrs.get_int_tuple("func_input_locs")
+    func_var_locs = attrs.get_int_tuple("func_var_locs")
+    num_out_data = attrs.get_int("num_out_data")
+    num_outputs = attrs.get_int("num_outputs")
+
+    import pdb; pdb.set_trace()
```

Comment:
do we need to keep them? same to the below ones.

Reply:
No, this was corrected in local commit will follow up